src/org/sosy_lab/java_smt/solvers/SolverLess/SolverLessFloatingPointFormulaManager.java

```diff
@@ -0,0 +1,242 @@
+    super(creator);
+  }
+
+  @Override
+  protected SMTLIB2Formula getDefaultRoundingMode() {
+    return new SMTLIB2Formula(new DummyType(FloatingPointRoundingMode.NEAREST_TIES_TO_EVEN));
+  }
+
+  @Override
+  protected SMTLIB2Formula getRoundingModeImpl(
+      FloatingPointRoundingMode pFloatingPointRoundingMode) {
+    return new SMTLIB2Formula(new DummyType(pFloatingPointRoundingMode));
+  }
+
+  @Override
+  protected SMTLIB2Formula makeNumberImpl(
+      double pN, FloatingPointType pType, SMTLIB2Formula pFloatingPointRoundingMode) {
+    return new SMTLIB2Formula(
+        new DummyType(
+            pType.getExponentSize(),
+            pType.getMantissaSize(),
+            pFloatingPointRoundingMode.getFormulaType().getRoundingMode()));
+  }
+
+  @Override
+  protected SMTLIB2Formula makeNumberImpl(
+      BigInteger exponent, BigInteger mantissa, boolean signBit, FloatingPointType type) {
+    return new SMTLIB2Formula(new DummyType(type.getExponentSize(), type.getMantissaSize()));
+  }
+
+  @Override
+  protected SMTLIB2Formula makeNumberAndRound(
+      String pN, FloatingPointType pType, SMTLIB2Formula pFloatingPointRoundingMode) {
+    return new SMTLIB2Formula(new DummyType(pType.getExponentSize(), pType.getMantissaSize()));
+  }
+
+  @Override
+  protected SMTLIB2Formula makePlusInfinityImpl(FloatingPointType pType) {
+    return new SMTLIB2Formula(pType.getExponentSize(), pType.getMantissaSize());
+  }
+
+  @Override
+  protected SMTLIB2Formula makeMinusInfinityImpl(FloatingPointType pType) {
+
+    return new SMTLIB2Formula(pType.getExponentSize(), pType.getMantissaSize());
+  }
+
+  @Override
+  protected SMTLIB2Formula makeNaNImpl(FloatingPointType pType) {
+    return new SMTLIB2Formula(pType.getExponentSize(), pType.getMantissaSize());
+  }
+
+  @Override
+  protected SMTLIB2Formula makeVariableImpl(String pVar, FloatingPointType pType) {
+    SMTLIB2Formula formula = new SMTLIB2Formula(pType.getExponentSize(), pType.getMantissaSize());
+    formula.setName(pVar);
+    return formula;
+  }
+
+  // Rest of the methods remain unchanged...
+  @Override
+  protected SMTLIB2Formula castToImpl(
+      SMTLIB2Formula pNumber,
+      boolean pSigned,
+      FormulaType<?> pTargetType,
+      SMTLIB2Formula pRoundingMode) {
+    if (pTargetType.isFloatingPointType()) {
+      FloatingPointType targetType = (FloatingPointType) pTargetType;
+      return new SMTLIB2Formula(targetType.getExponentSize(), targetType.getMantissaSize());
+    }
+    if (pTargetType.isIntegerType()) {
+      return new SMTLIB2Formula(new DummyType(DummyType.Type.INTEGER));
+    }
+    if (pTargetType.isBitvectorType()) {
+      BitvectorType targetType = (BitvectorType) pTargetType;
+      return new SMTLIB2Formula(targetType.getSize());
+    }
+    if (pTargetType.isRationalType()) {
+      return new SMTLIB2Formula(new DummyType(DummyType.Type.RATIONAL));
+    }
+    return null;
```

Comment:
Never return plain `null` for known reasons.
This looks like a switch-case expression.